src/App.tsx

```diff
@@ -96,8 +98,14 @@ const App = () => {
             <Route path='feed' element={<FeedClaim {...commonProps} />} />
             <Route path='search' element={<Search {...commonProps} />} />
             <Route path='/' element={<Form {...commonProps} />} />
-            <Route path='login' element={<Login {...commonProps} />} />
             <Route path='register' element={<Register {...commonProps} />} />
+            <Route path='login' element={<Login {...commonProps} />} />
+            <Route
+              path='/rate'
+              element={
+                checkAuth() ? <Rate {...commonProps} /> : <Navigate to='/login' replace state={{ from: location }} />
```

Comment:
Make a common function that is reusable

Reply:
this is also good idea!  @sahdasamier @Zakiamangal if you want to take this